File: project10/ruby/Bird.rb
# Bird.rb | Defines a Bird superclass 
#   to be extended by specific bird sub-classes.
#
# Completed by: Haim Hong
# Date: March 25 2022
####################################################

class Bird
    attr_reader :name

    def initialize(name) # initializes it
        @name = name
    end

    def call # call function for this class specific
        'Squaaaaaaawk!'
    end

    def className # use the class name for what the animal is
        self.class.to_s
    end
    
    def movement
        'flews'
    end

    def print # pritns out 
        puts name + ' ' + className + " just " + movement + " and said " + call
    end
end

File: project10/java/Bird.java
/* Bird.java provides a Bird class.
 *
 * Completed by: Haim Hong
 * Date: March 25 2022
 ******************************************************/

import java.io.*;

public class Bird
{

 /* default constructor
  * PostCond: myName == "".
  */
  public Bird(){
    myName = "";
  }

 /* explicit constructor
  * Receive: name, a String 
  * PostCond: myName == name.
  */
  public Bird(String name){
    myName = name;
  }

 /* Name accessor
  * Return: myName.
  */
  public String getName(){
    return myName;
  }

 /* A Bird's Call
  * Return: a default bird-call ("Squawk!").
  */
  public String call()
  {
    return "Squaaaaaaaaaawk!";
  }
/* A Bird's movement
 *  Return: a default bird moement ("flew")
 */
  public String movement(){
    return "flew";
  }

 /* Output a Bird 
  * Output: Everything known about myself
  *          to the standard output stream.
  */
  public void print()
  {
    System.out.println(getName() + ' ' + getClass().getName() + " just " + movement() + " and said " + call());
  }


  private String myName;
}

